internal_packages/keybase/lib/pgp-key-store.cjsx

```diff
@@ -0,0 +1,391 @@
+NylasStore = require 'nylas-store'
+{DraftStore, MessageBodyProcessor, RegExpUtils} = require 'nylas-exports'
+{remote} = require 'electron'
+kb = require './keybase'
+pgp = require 'kbpgp'
+_ = require 'underscore'
+path = require 'path'
+fs = require 'fs'
+
+class PGPKeyStore extends NylasStore
+
+  constructor: ->
+    super()
+
+    @_pubKeys = {}
+    @_privKeys = []
+
+    @_msgCache = []
+    @_msgStatus = []
+
+    @_pubWatcher = null
+    @_privWatcher = null
+
+    @_keyDir = path.join(NylasEnv.getConfigDirPath(), 'keys')
+    @_pubKeyDir = path.join(@_keyDir, 'public')
+    @_privKeyDir = path.join(@_keyDir, 'private')
+
+    # Create the key storage file system if it doesn't already exist
+    fs.access(@_keyDir, fs.R_OK | fs.W_OK, (err) =>
+      if err
+        fs.mkdir(@_keyDir, (err) =>
+          if err
+            console.warn err
+          else
+            fs.mkdir(@_pubKeyDir, (err) =>
+              if err
+                console.warn err
+              else
+                fs.mkdir(@_privKeyDir, (err) =>
+                  if err
+                    console.warn err
+                  else
+                    @watch())))
+      else
+        fs.access(@_pubKeyDir, fs.R_OK | fs.W_OK, (err) =>
+          if err
+            fs.mkdir(@_pubKeyDir, (err) =>
+              if err
+                console.warn err))
+        fs.access(@_privKeyDir, fs.R_OK | fs.W_OK, (err) =>
+          if err
+            fs.mkdir(@_privKeyDir, (err) =>
+              if err
+                console.warn err))
+        @_populate(isPub = true)
```

Comment:
Yes, that's why I'm doing that, and it's always seemed to work - although I admit I'm much more familiar with Python than CoffeeScript. I think the actually correct way to do this is with the expanded calls that assign inputs as object attributes - so instead of `isPub=true`, this'd be `{isPub: true}`, but then you gotta be careful to change it everywhere.